server/pulp/server/db/manage.py

```diff
@@ -191,6 +196,34 @@ def main():
         options = parse_args()
         _start_logging()
         connection.initialize(max_timeout=1)
+
+        if status.get_workers():
+            last_worker_time = max([worker['last_heartbeat'] for worker
+                                    in status.get_workers()])
+            wait_time = timedelta(seconds=constants.MIGRATION_WAIT_TIME) -\
+                (UTCDateTimeField().to_python(datetime.now()) - last_worker_time)
+
+            if wait_time > timedelta(0):
+                print _('The following processes might still be running:')
+                for worker in status.get_workers():
+                    print _('\t%s' % worker['name'])
+
+                for i in range(wait_time.seconds, 0, -1):
+                    print _('\rPlease wait %s seconds while Pulp confirms this.' % i),
+                    sys.stdout.flush()
+                    time.sleep(1)
+
+                if status.get_workers() and \
+                        max([worker['last_heartbeat'] for worker in
```

Comment:
Since we're filtering workers, which there will never be _that_ many of, I think moving the filter to the database would not be too much more efficient.